spec/haskell/src/SEL4/Object/SchedContext.lhs

```diff
@@ -643,7 +643,8 @@ This module uses the C preprocessor to select a target architecture.
 >     when (numDomains > 1) $ do
 >       consumed <- getConsumedTime
 >       domainTime <- getDomainTime
->       setDomainTime (domainTime - consumed)
+>       time' <- return $ if (domainTime < consumed) then 0 else (domainTime - consumed)
```

Comment:
I may not have kept up with what's going on here. The C currently doesn't appear to clip this at zero. Is there a C change in the pipeline?

Reply:
Mmmm, not yet. I suppose I should do it all at once. Good point.

Reply:
This is now in seL4 master branch. Thanks for prompting this.